team-for-linux: 1.3.2 -> 1.3.8

qjoly opened the pull request:
## Description of changes
Ful changelog: https://github.com/IsmaelMartinez/teams-for-linux/compare/v1.3.2...v1.3.5

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

lilyinstarlight commented on pkgs/applications/networking/instant-messengers/teams-for-linux/default.nix:
```diff
@@ -81,6 +81,11 @@ stdenv.mkDerivation (finalAttrs: {
     runHook postInstall
   '';
 
+  doInstallCheck = true;
+  installCheckPhase = ''
+    $out/bin/teams-for-linux --version | grep -F "${version}" >/dev/null
+  '';
+
```
Should we just use `testers.testVersion` instead if we really want to add this?

```suggestion
  passthru.tests.version = testers.testVersion { package = teams-for-linux; };

```

lilyinstarlight commented on pkgs/applications/networking/instant-messengers/teams-for-linux/default.nix:
```diff
@@ -15,15 +15,15 @@
 , which
```
(to support above review item)
```suggestion
, which
, testers
, teams-for-linux
```

lilyinstarlight commented on pkgs/applications/networking/instant-messengers/teams-for-linux/default.nix:
```diff
@@ -15,15 +15,15 @@
 }:
 
-stdenv.mkDerivation (finalAttrs: {
+stdenv.mkDerivation (finalAttrs: rec {
```
```suggestion
stdenv.mkDerivation (finalAttrs: {
```

We are using the `finalAttrs` pattern instead of `rec { ... }`

lilyinstarlight commented:
Teams-for-linux just released 1.3.7, if you're able to find time to update this PR. Thank you for the contribution!

QJoly commented:
> Teams-for-linux just released 1.3.7, if you're able to find time to update this PR. Thank you for the contribution!

I can update it Monday or Tuesday, I'm currently on vacation.

QJoly commented:
Thank you for your code-review, 

I updated the code using your suggestions :)

lilyinstarlight commented on pkgs/applications/networking/instant-messengers/teams-for-linux/default.nix:
```diff
@@ -13,17 +13,19 @@
 }:
 
-stdenv.mkDerivation (finalAttrs: {
+stdenv.mkDerivation (finalAttrs: rec {
```
```suggestion
stdenv.mkDerivation (finalAttrs: {
```

Can you still take out this `rec`? Otherwise this looks great now, thank you!

qjoly committed:
teams-for-linux: 1.3.2 -> 1.3.8

Update pkgs/applications/networking/instant-messengers/teams-for-linux/default.nix

Co-authored-by: Lily Foster <lily@lily.flowers>

lilyinstarlight commented on pkgs/applications/networking/instant-messengers/teams-for-linux/default.nix:
```diff
@@ -91,12 +93,13 @@ stdenv.mkDerivation (finalAttrs: {
   })];
 
   passthru.updateScript = ./update.sh;
+  passthru.tests.version = testers.testVersion { package = teams-for-linux; };
```
```suggestion
  passthru.tests.version = testers.testVersion rec {
    package = teams-for-linux;
    command = "HOME=$TMPDIR ${package.meta.mainProgram or package.pname} --version";
  };
```

Okay so it turns out doing `teams-for-linux --version` expects a writable `HOME` set to be able to work and currently it fails

So this diff should do it, and then this PR is good to go. Apologies for the back and forth, but thank you again for working on this!

lilyinstarlight commented:
> testing this doesn't resolve [#249081 (comment)](https://github.com/NixOS/nixpkgs/pull/249081#issuecomment-1707732925). I'm not sure if this is a widespread problem but it breaks teams entirely on my installation.

I don't experience this on my system, but upstream did downgrade their own electron to 24 iirc after upgrading gave some problems. So we probably should use `electron_24` for now, especially if it fixes things for you

lilyinstarlight commented:
> testing this doesn't resolve [#249081 (comment)](https://github.com/NixOS/nixpkgs/pull/249081#issuecomment-1707732925). I'm not sure if this is a widespread problem but it breaks teams entirely on my installation.

I _think_ it should merge cleanly with this, if you want to make a separate PR that modifies `electron` to `electron_24` in `pkgs/applications/networking/instant-messengers/teams-for-linux/default.nix` in the mean time

Thank you for bring up that issue!

dschrempf commented:
FYI, I am also having trouble with the version of Teams currently on `nixos-unstable` (not this one). Changing electron to `electron_24` fixes the issues, as described in https://github.com/NixOS/nixpkgs/pull/249081.

lilyinstarlight commented:
Downgrade to `electron_24` is at #254379, so hopefully it doesn't make a merge conflict with this PR when it's merged

@QJoly if you have time, that one suggested change above is all that should be needed to get this merged. If you don't have time, I should be able to push to this branch myself if you'd like or I'm just happy to wait. Regardless, thank you for this work! :)

lilyinstarlight reviewed (approved):
I'm waiting on ofborg now, but this looks good to me. Thanks again!
